cpp/include/cudf/types.hpp

```diff
@@ -126,7 +126,7 @@ struct order_info {
 /**---------------------------------------------------------------------------*
  * @brief Controls the allocation/initialization of a null mask.
  *---------------------------------------------------------------------------**/
-enum mask_state {
+enum class mask_state : int {
```

Comment:
Do we need to explicitly make this an int? If so, why not an `int8_t`? None of our other enum classes do this, unless they are binary in which case we use `bool`. @jrhemstad opinions?

Reply:
So this came from an offline conversation with @shwina and @jrhemstad where C++ enum classes are `int` if they're not explicitly typed. I'm perfectly happy making this an `int8_t` and defer to your judgement here.

For reference: https://en.cppreference.com/w/cpp/language/enum#Scoped_enumerations

Reply:
Making the type explicit makes it safer and more convenient for interfacing with Cython. Otherwise they have to rely on the implicit behavior of an untyped `enum class` being an `int`. This way it is just more explicit.

Reply:
@jrhemstad @harrism so should we change this to `int8_t` instead of `int`? Seems like being explicit is a good thing here regardless though 😄.

Reply:
I don't know about Cython interfacing, but if we want to be concrete with the type, (u)int32_t seems like a better option than int.

Reply:
Cython is perfectly happy to use whatever type as long as we explicitly define the type so that we're not relying on language / compiler implicit behavior.

I defer the type decision to someone on the cpp side 😄

Reply:
The only reason to use int8_t is if we a) are worried about storage / bandwidth and b) we are sure we will never need more than 128 values in the enum.

Since we probably won't have columns of enum cvalues, `int32_t` should be fine, and use it on all non-`bool` enum classes.